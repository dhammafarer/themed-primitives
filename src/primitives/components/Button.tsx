import * as React from "react";
import { css, styled, Scale } from "../../theme";
import { Box, box, BoxProps, ResponsiveScale, ResponsiveString } from "./Box";
import {
  getBoxShadow,
  getBorder,
  getBorderRadius,
  getBorderColor,
  getFontFamily,
  getFontSize,
  getFontWeight,
  getLineHeight,
  getLetterSpacing,
  getTextTransform,
  getTextAlign,
} from "../getters";

interface ButtonProps extends BoxProps {
  shadow?: ResponsiveScale;
  radius?: ResponsiveScale;
  b?: ResponsiveScale;
  borderColor?: ResponsiveString;
  fontFamily?: string;
  fontSize?: ResponsiveScale;
  fontWeight?: ResponsiveScale;
  textTransform?: ResponsiveString;
  textAlign?: ResponsiveString;
  lineHeight?: ResponsiveString;
  letterSpacing?: ResponsiveString;
}

const button = css<ButtonProps>`
  text-decoration: none;
  cursor: pointer;
  ${box}
  ${props => css`
    ${getBoxShadow(props)}
    ${getBorder(props)}
    ${getBorderColor(props)}
    ${getBorderRadius(props)}
    ${getFontSize(props)}
    ${getFontWeight(props)}
    ${getFontFamily(props)}
    ${getLineHeight(props)}
    ${getLetterSpacing(props)}
    ${getTextTransform(props)}
    ${getTextAlign(props)}
  `}
`;

const Button = styled(
  ({
    shadow,
    b,
    bx,
    by,
    bt,
    bb,
    br,
    bl,
    borderColor,
    radius,
    fontSize,
    fontWeight,
    fontFamily,
    lineHeight,
    letterSpacing,
    textTransform,
    textAlign,
    ...rest
  }) => <Box {...rest} />
)<ButtonProps & React.HTMLProps<HTMLButtonElement>>`
  ${button}
`;

export { Button, ButtonProps };
