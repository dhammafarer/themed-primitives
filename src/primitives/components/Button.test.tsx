import * as React from "react";
import * as renderer from "react-test-renderer";

import { Button } from "./Button";

test("should render a div", () => {
  const tree = renderer.create(<Button />).toJSON();

  expect(tree).toMatchSnapshot();
});
