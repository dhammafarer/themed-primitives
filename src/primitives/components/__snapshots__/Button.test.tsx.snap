// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a div 1`] = `
<div
  className="sc-bwzfXH dLnHtn sc-bdVaJa jolpEI"
/>
`;
